src/PowerShellEditorServices/Language/FindFoldsVisitor.cs

```diff
@@ -0,0 +1,142 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+
+namespace Microsoft.PowerShell.EditorServices
+{
+    /// <summary>
+    /// The visitor used to find the all folding regions in an AST
+    /// </summary>
+    internal class FindFoldsVisitor : AstVisitor
```

Comment:
Sorry if I sound like a broken record, but we're sure this works with classes right?  I would have expected this to either need to inherit `AstVisitor2` or dip into tokens for classes.  Can we get a basic class example added to the tests? Something like

```powershell
class TestClass {
    [string[]] $TestProperty = @(
        'first',
        'second',
        'third')

    [string] TestMethod() {
        return $this.TestProperty[0]
    }
}
```

Reply:
Welp...this is a problem....works on master but not after this PR.... More work for me to do.

Reply:
So we did indeed need to use ASTVisitor2 for the `VisitTypeDefinition` method.  Added a test for this and is now passing.

Fixed.